execution-engine/src/wasi/common.rs

```diff
@@ -150,12 +152,15 @@ impl TryFrom<&str> for WasiAPIName {
 ////////////////////////////////////////////////////////////////////////////////
 
 /// Unpack a sequence of `bytes` and return a `T`.
-pub trait Unpack<T> {
-    fn unpack(bytes: &[u8]) -> FileSystemResult<T>;
+pub trait Unpack: Sized {
+    const SIZE: u32;
```

Comment:
What is the purpose of this `SIZE`?  Can we get `rustdoc` comments on the methods/associated constants of the trait?

Reply:
Ah yes can do. Previously the Unpack trait relied on size_of to determine the size to read from the linear memory. Having a separate associated constant means you can unpack things that have different sizes in/outside of Wasm.